### ISSUE TITLE
Fennec migration: The Release migration build number is displayed as Beta

### ISSUE THREAD

**Issue by ebalazs-sv** [mozilla-mobile/android-components#5262]
## Preconditions:
Have Firefox Release installed and logged in.
## Steps to reproduce
1. Replace Fennec with Fenix using a build from https://github.com/mozilla-mobile/fenix/wiki/Fennec-Migration;
2. Launch Fenix;
3. Go to Settings -> About Firefox Preview and observe the build number.
### Expected behavior
The build number is displayed as "Nightly 191209 6:07".
### Actual behavior
The build number is displayed as "null".
### Notes:
[Screenshot](https://drive.google.com/file/d/1bilyEqf2lVwtnsUcMdnCnJ9C9W4dIvdi/view)
### Device information

* Android device: Google Pixel 2 (Android 9) - ARM64; Nexus 6P (Android 8.1.0) - ARM; Xiaomi Mi Pad 2 (Android 5.1) - x86.
* Fenix version: Nightly 191209 GV 71

**Comment by Sparky93** [comment]
I would like to work on this.

**Comment by ebalazs-sv** [comment]
I tested this on today's Release, Beta, Nightly migration build 12/18 with Google Pixel 2 (Android 9), Samsung Galaxy S8 (Android 9) - ARM64; Nexus 6P (Android 8.1.0) - ARM; Xiaomi Mi Pad 2 (Android 5.1) - x86 and the build number is displayed as follows:
- Release: 3.1.0-beta.1 GV 72;
- Beta: 3.1.0-beta.1 GV 72;
- Nightly: Nightly 191218 GV 73;

**Comment by ebalazs-sv** [comment]
I had tested this on Release, Beta, Nightly migration builds from 01/10 with Samsung Galaxy S8 (Android 9), Huawei Honor 8 (Android 7.0), Xiaomi Mi4i (Android 5.0.2) and the build numbers are displayed as follows:
- Release: 3.1.0-beta.1 GV 73;
- Beta: 3.1.0-beta.1 GV 73;
- Nightly: Nightly 200110 GV 74;

Due to the different behavior I will change the bug title.

**Comment by Sparky93** [comment]
Hello, it looks like the last behavior is intended, based on this [line](https://github.com/mozilla-mobile/fenix/blob/master/app/build.gradle#L113) so I'd propose to close this as invalid, @liuche how do you suggest to proceed further?

**Comment by liuche** [comment]
@st3fan ^^

**Comment by pocmo** [comment]
@JohanLorenzo Since we pass the version string to the build in automation: Do you know why the `fennecProduction` build picks up a beta version string? Is this only because of our "build this nightly" setup or do you think that will also affect our actual builds once we trigger them through a tagged release?

**Comment by JohanLorenzo** [comment]
Hi there! 

**TL;DR:** `fennecProduction` contains a bug I didn't foresee when landing https://github.com/mozilla-mobile/fenix/pull/7081. To me, this bug looks minor and shouldn't impact the real migration because we will ship releases thanks to Github Releases. Anyway, Releng should own the fix. I see 4 ways of doing so. Please let me know what you think @Callek. 

Thanks for looping me in 🙂

From what I understand, this ticket covers 2 different issues:

1. `About Firefox Preview` displays a `null` version.
2. `fennecProduction` displays a beta version number.

I don't know about the first item, although I can answer regarding the second one.

`fennecProduction` is currently part of the [nightly graph](https://github.com/mozilla-mobile/fenix/blob/941c1479893becd7d59a04a1130f09f73bef996f/taskcluster/ci/build/kind.yml#L150-L155) because we need to qualify the migration early. This means there is no Github-Release created which means there is no tag to specify the version number.

We introduced a [version file](https://github.com/mozilla-mobile/fenix/blob/941c1479893becd7d59a04a1130f09f73bef996f/version.txt) stored in the repo in https://github.com/mozilla-mobile/fenix/pull/7081. This version file is meant to replace Github-Releases in order to enable a Releng's tool called [Shipit](https://shipit.mozilla-releng.net/). Shipit will make releases more secure than when kicked off regular Github-Releases. We intend to fully enable once the Fennec migration is done. This version file will be automatically bumped by Shipit once a realease is officially out. At the moment, this file is present for testing only while Releng keeps enabling Shipit. 

Anyway, this version file is the default version is [there is no tag specified](https://github.com/mozilla-mobile/fenix/blob/941c1479893becd7d59a04a1130f09f73bef996f/taskcluster/fenix_taskgraph/__init__.py#L41-L43) - which is what happens in the nightly graph. That's why `fennecProduction` ends up with this [beta version number](https://github.com/mozilla-mobile/fenix/blob/941c1479893becd7d59a04a1130f09f73bef996f/taskcluster/fenix_taskgraph/transforms/build.py#L96).

I see 4 ways of solving this bug:
 a. Either we move `fennecProduction` off nightly and the bug will go away.
 b. Or we revert https://github.com/mozilla-mobile/fenix/pull/7081.
 c. Or we don't set `parameters["release_version"]` in the nightly graph so no version number is defined.
 d. Or we define 2 different version files, just like Fennec does: [beta file](https://hg.mozilla.org/releases/mozilla-esr68/file/8eec70a45ee0dee552ae8e01a17c2623d9834c6f/mobile/android/config/version-files/beta/version_display.txt), [release file](https://hg.mozilla.org/releases/mozilla-esr68/file/8eec70a45ee0dee552ae8e01a17c2623d9834c6f/mobile/android/config/version-files/release/version_display.txt)

I would be inclined to go for option d. @Callek, what do you think?

**Comment by pocmo** [comment]
> To me, this bug looks minor and shouldn't impact the real migration because we will ship releases thanks to Github Release

If it does not impact real releases then I think it is absolutely okay to just close the issue. We will pass an "actual release" to QA before starting beta/production migration and can at that point verify the version. I think there is no need to fix that for our "nightly" setup.

**Comment by pocmo** [comment]
@ebalazs-sv I am going to close this issue and move it to the "QA" column. However note that this will not be fixed until we officially tag a build for sign-off.

**Comment by ebalazs-sv** [comment]
We will check this with the official Release Migration build. 
I will move this to Done. Thanks!

**Comment by Callek** [comment]
@JohanLorenzo if we were to tackle from code, (d) probably makes the most sense, followed by (c). I'm happy with the (e) [close this issue] too.
